src/Core/Grid/Definition/Factory/MerchandiseReturnProductsGridDefinitionFactory.php

```diff
@@ -0,0 +1,255 @@
+use PrestaShop\PrestaShop\Core\Grid\Filter\Filter;
+use PrestaShop\PrestaShop\Core\Grid\Filter\FilterCollection;
+use PrestaShop\PrestaShop\Core\Hook\HookDispatcherInterface;
+use PrestaShopBundle\Form\Admin\Type\SearchAndResetType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * Class MerchandiseReturnGridDefinitionFactory builds grid definition for merchandise returns grid.
+ */
+final class MerchandiseReturnProductsGridDefinitionFactory extends AbstractFilterableGridDefinitionFactory
+{
+    use BulkDeleteActionTrait;
+    use DeleteActionTrait;
+
+    const GRID_ID = 'merchandise_return_products';
+
+    /**
+     * @var int
+     */
+    private $merchandiseReturnId;
+
+    /**
+     * MerchandiseReturnProductsGridDefinitionFactory constructor.
+     *
+     * @param HookDispatcherInterface $hookDispatcher
+     * @param RequestStack $requestStack
+     */
+    public function __construct(
+        HookDispatcherInterface $hookDispatcher,
+        RequestStack $requestStack
+    ) {
+        parent::__construct($hookDispatcher);
+        $this->setMerchandiseReturnId($requestStack);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getId()
+    {
+        return self::GRID_ID;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getName()
+    {
+        return $this->trans('Merchandise return products', [], 'Admin.Orderscustomers.Feature');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getColumns()
+    {
+        $columns = (new ColumnCollection())
+            ->add(
+                (new BulkActionColumn('merchandise_return_bulk'))
+                    ->setOptions([
+                        'bulk_field' => 'id_order_detail',
+                        'extra_fields' => [
+                            'id_customization',
+                        ],
+                    ])
+            )
+            ->add(
+                (new DataColumn('product_reference'))
+                    ->setName($this->trans('Reference', [], 'Admin.Global'))
+                    ->setOptions([
+                        'field' => 'product_reference',
+                    ])
+            )
+            ->add(
+                (new DataColumn('product_name'))
+                    ->setName($this->trans('Product name', [], 'Admin.Global'))
+                    ->setOptions([
+                        'field' => 'product_name',
+                    ])
+            )
+            ->add(
+                (new DataColumn('quantity'))
+                    ->setName($this->trans('Quantity', [], 'Admin.Global'))
+                    ->setOptions([
+                        'field' => 'product_quantity',
+                    ])
+            )
+            ->add(
+                (new MerchandiseReturnCustomizationColumn('customizations'))
+                    ->setName($this->trans('Customizations', [], 'Admin.Global'))
```

Comment:
Hey, answered here https://github.com/PrestaShop/PrestaShop/pull/19420#discussion_r450177943

Reply:
Oh, thank you very much @JevgenijVisockij! @Junebyun, I think you should decide whether or not we keep the plural form here. I don't know if it is very common to have a plural form as a grid feature. :-)

Reply:
The reason I decided to make it plural is because each product can have multiple customizations in one row. I am not sure if that changes anything for you.

Reply:
Yep, I saw that on the screenshot below but it is more like a question of terminology and consistency: column labels are usually written in a more general sense and therefore at the singular form - but perhaps it is different in the software. @Junebyun will tell us. :-)

Reply:
I checked the BO and there were some instances where plural was used for grid features. Since each product can have multiple customizations I'm okay with keeping "_Customizations_" in plural! :)